seismicpro/src/utils.py

```diff
@@ -589,17 +591,25 @@ def make_segy_index(filename, extra_headers=None, limits=None):
         segyfile.mmap()
         if extra_headers == 'all':
             headers = [h.__str__() for h in segyio.TraceField.enums()]
+            del_headers = []
         elif extra_headers is None:
-            headers = DEFAULT_SEGY_HEADERS
+            headers = DEFAULT_SEGY_HEADERS + SUPPORT_SEGY_HEADERS
+            del_headers = SUPPORT_SEGY_HEADERS
         else:
-            headers = set(DEFAULT_SEGY_HEADERS + list(extra_headers))
+            extra_headers = extra_headers if isinstance(extra_headers, (list, tuple)) else [extra_headers]
```

Comment:
array-likes are converted to lists, so that `+` op works in the next line
```suggestion
            extra_headers = [extra_headers] if isinstance(extra_headers, str) else list(extra_headers)
```